elide-core/src/main/java/com/yahoo/elide/jsonapi/models/Relationship.java

```diff
@@ -80,6 +81,8 @@ public Relationship(@JsonProperty("links") Map<String, String> links,
                     res.add(resource.toPersistentResource(requestScope));
                 } catch (ForbiddenAccessException e) {
                     //skip resource
+                } catch (java.lang.NullPointerException e) {
+                    throw new InvalidRelationException(resource.getType());
```

Comment:
I'm guessing that the exception is ultimately being thrown [from here](https://github.com/yahoo/elide/blob/65929a6b13e1798afd7ba06b12775e660803881c/elide-core/src/main/java/com/yahoo/elide/jsonapi/models/Resource.java#L140) and so we should throw this `InvalidRelationshipException` there.